src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -92,11 +102,35 @@ public String getWsmServiceAccountToken() throws IOException {
     return creds.getAccessToken().getTokenValue();
   }
 
+  @FunctionalInterface
+  public interface InterruptedSupplier<R> {
+    R apply() throws InterruptedException;
+  }
+
+  public interface VoidInterruptedSupplier {
```

Comment:
worth some javadoc tying these to the rethrowIfSamInterrupted

Reply:
Done.